### PR TITLE
Buy modal

### PR DESCRIPTION
**Pull request by kingbrendann**

### CODE REVIEW

File: More2Life.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -981,7 +1025,7 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Distribution";
```
Comment (PorterHoskins) [inline More2Life.xcodeproj/project.pbxproj]: Not sure we want this.

File: More2Life.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -991,8 +1035,8 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks";
 				PRODUCT_BUNDLE_IDENTIFIER = com.coachkalani.More2Life;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE = "a5a22912-8745-4c07-87e1-461f297c4457";
```
Comment (PorterHoskins) [inline More2Life.xcodeproj/project.pbxproj]: not sure about this...

File: More2Life.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -991,8 +1035,8 @@
...
-				PROVISIONING_PROFILE_SPECIFIER = "match AppStore com.coachkalani.More2Life";
```
Comment (PorterHoskins) [inline More2Life.xcodeproj/project.pbxproj]: I think i had this setup right before...

File: More2Life/HalfModalTransitioningDelegate.swift [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        super.init()
+    }
+	
+    func presentationController(forPresented presented: UIViewController, presenting: UIViewController?, source: UIViewController) -> UIPresentationController? {
+        return HalfModalPresentationController(presentedViewController: presented, presenting: presenting)
+    }
+	
+}
+
+extension UIViewController { }
```
Comment (PorterHoskins) [inline More2Life/HalfModalTransitioningDelegate.swift]: ??

File: More2Life/MainButton.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+//  More2Life
+//
+//  Created by Brendan Kingsford on 8/29/17.
+//  Copyright © 2017 More2Life. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class MainButton: UIButton {
```
Comment (PorterHoskins) [inline More2Life/MainButton.swift]: If you want this in a storyboard with @IBDesignable then you will want to put this in the Shared framework.

File: More2Life/PriceButton.swift [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+//  More2Life
+//
+//  Created by Brendan Kingsford on 8/29/17.
+//  Copyright © 2017 More2Life. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class PriceButton: UIButton {
```
Comment (PorterHoskins) [inline More2Life/PriceButton.swift]: If you want this in a storyboard with @ibdesignable then you will want to put this in the Shared framework.

**Review by PorterHoskins — CHANGES_REQUESTED** [review]
Just change the provisioning back on the distribution.

File: More2Life/PriceButton.swift (hunk 6 above)
Comment (PorterHoskins) [reply More2Life/PriceButton.swift]: Putting @IBDesignable on a class inside the main project requires the storyboard to build the entire project to render the storyboard. Not good.

Comment (PorterHoskins) [reply More2Life/PriceButton.swift]: If you don't want IBDesignable then just ignore this.

File: More2Life.xcodeproj/project.pbxproj (hunk 3 above)
Comment (kingbrendann) [reply More2Life.xcodeproj/project.pbxproj]: oh i see. match lets you choose `development, adhoc, enterprise  or appstore` and you are using the appstore profile. fixed.

**Comment by kingbrendann** [comment]
Ok all that is fixed.

File: More2Life/DetailViewController.swift [hunk 7]
```diff
@@ -104,6 +106,16 @@ class DetailViewController: UIViewController, FeedDetailing {
...
+		case let feedItem as ListingFeedItem:
+			let storyboard = UIStoryboard(name: "Main", bundle: nil)
+			let buyModalViewController = storyboard.instantiateViewController(withIdentifier: "BuyModalViewController")
+			
+			self.halfModalTransitioningDelegate = HalfModalTransitioningDelegate(viewController: self, presentingViewController: buyModalViewController)
+			
+			buyModalViewController.modalPresentationStyle = .custom
+			buyModalViewController.transitioningDelegate = self.halfModalTransitioningDelegate
+			
+			present(buyModalViewController, animated: true, completion: nil);
```
Comment (PorterHoskins) [inline More2Life/DetailViewController.swift]: semi-colon 😱

File: More2Life/FeedViewController.swift [hunk 8]
```diff
@@ -87,6 +89,18 @@ class FeedViewController: UIViewController {
...
+	@IBAction func buyTapped(_ sender: Any) {
+		let storyboard = UIStoryboard(name: "Main", bundle: nil)
+		let buyModalViewController = storyboard.instantiateViewController(withIdentifier: "BuyModalViewController")
+		
+		self.halfModalTransitioningDelegate = HalfModalTransitioningDelegate(viewController: self, presentingViewController: buyModalViewController)
+		
+		buyModalViewController.modalPresentationStyle = .custom
+		buyModalViewController.transitioningDelegate = self.halfModalTransitioningDelegate
+		
+		present(buyModalViewController, animated: true, completion: nil);
```
Comment (PorterHoskins) [inline More2Life/FeedViewController.swift]: semi-colon 😱

File: More2Life/HalfModalPresentationController.swift [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+        // Vibrancy Effect
+        let vibrancyEffect = UIVibrancyEffect(blurEffect: blurEffect)
+        let vibrancyEffectView = UIVisualEffectView(effect: vibrancyEffect)
+        vibrancyEffectView.frame = view.bounds
+        
+        // Add the vibrancy view to the blur view
+        blurEffectView.contentView.addSubview(vibrancyEffectView)
+		
+		let gestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissOnTap))
+		view.addGestureRecognizer(gestureRecognizer);
```
Comment (PorterHoskins) [inline More2Life/HalfModalPresentationController.swift]: semi-colon 😱

File: More2Life/HalfModalPresentationController.swift [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+		
+        return view
+    }
+	
+	func dismissOnTap(sender: UITapGestureRecognizer) {
+		presentingViewController.dismiss(animated: true, completion: nil)
+	}
+ 
+    override var frameOfPresentedViewInContainerView: CGRect {
+        return CGRect(x: 0, y: containerView!.bounds.height * 0.333 , width: containerView!.bounds.width, height: containerView!.bounds.height * 0.666)
```
Comment (PorterHoskins) [inline More2Life/HalfModalPresentationController.swift]: `!` is also known as the crash-operator. If this view doesn't have a content view then it is probably right to crash but usually I like to do something like:

```
guard let containerView = containerView else {
    fatalError("Justify the crash so everyone knows the reason why it is ok to crash.")
}
```

Comment (PorterHoskins) [reply More2Life/HalfModalPresentationController.swift]: Other option would be to do:

`containerView?.bounds.width ?? 0`

**Review by PorterHoskins — CHANGES_REQUESTED** [review]

**Review by PorterHoskins — APPROVED** [review]

**Pull request merged**